open-amp/lib/virtio/virtqueue.c

```diff
@@ -241,15 +262,24 @@ void *virtqueue_get_available_buffer(struct virtqueue *vq, uint16_t *avail_idx,
 	void *buffer;
 
 	atomic_thread_fence(memory_order_seq_cst);
+
+	/* Avail.idx is updated by master, invalidate it */
+	VRING_INVALIDATE(vq->vq_ring.avail->idx);
 	if (vq->vq_available_idx == vq->vq_ring.avail->idx) {
 		return NULL;
 	}
 
 	VQUEUE_BUSY(vq);
 
 	head_idx = vq->vq_available_idx++ & (vq->vq_nentries - 1);
+
```

Comment:
Here there is a delta vs the original open-amp lib. The code below is placed before previous line.

Reply:
well, that's embarrassing. I had some local changes in my open-amp repo that were picked up when copying over. Sorry about this.